### PR TITLE
Add Datefield to application system

### PR DESCRIPTION
**Pull request by arnifreyrs**
## What

Create new library: `shared-form-fields` for components used in a "react hook form"-form. Moved all the controllers from island-ui into that new lib. Added `DatePickerController`

## Why

We need a datepicker in the application system.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against master before asking for a review

### CODE REVIEW

File: libs/shared/form-fields/src/lib/DatePickerController/DatePickerController.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          id="id"
+          label="date"
+          placeholder="great placeholder"
+        />
+      </Wrapper>,
+    )
+    expect(getByText('great placeholder')).toBeInTheDocument()
+  })
+
+  it('should render no placeholder if the daatepicker has a selected value', () => {
```
Comment (madebynoam) [inline libs/shared/form-fields/src/lib/DatePickerController/DatePickerController.spec.tsx:48]: minor, typo `daatepicker ` :)

File: apps/application-system/form/src/styles.scss [hunk 2]
```diff
@@ -0,0 +1,45 @@
+@font-face {
```
Comment (madebynoam) [inline apps/application-system/form/src/styles.scss:1]: Should these be part of core theme?

**Review by madebynoam — APPROVED** [review]
Lgtm!

File: apps/application-system/form/src/styles.scss (hunk 2 above)
Comment (arnifreyrs) [reply apps/application-system/form/src/styles.scss:1]: probably, I just don't know how to do it in another way :/ just followed Kolibri's lead on this.

**Review by madebynoam — APPROVED** [review]

**Review by pshomov — APPROVED** [review]
LGTM

File: apps/application-system/form/src/styles.scss (hunk 2 above)
Comment (jeremybarbet) [reply apps/application-system/form/src/styles.scss:1]: Agree, it should probably be moved to the island-ui. One scss file that could be imported along with the fonts folder

File: workspace.json [hunk 3]
```diff
@@ -1866,7 +1866,7 @@
               "apps/application-system/form/src/favicon.ico",
               "apps/application-system/form/src/assets"
             ],
-            "styles": [],
+            "styles": ["apps/application-system/form/src/styles.scss"],
```
Comment (jeremybarbet) [inline workspace.json:1869]: Could the logic be done here: `libs/island-ui/core/src/global/fonts.treat.ts`

and above
```
"assets" : [
  ...,
  libs/island-ui/core/src/assets/fonts
]
```

this is the theory, not sure if it works

**Review by jeremybarbet — APPROVED** [review]
LGTM, apart from the font question

File: workspace.json (hunk 3 above)
Comment (oddsson) [reply workspace.json:1869]: I believe there is an open issue on the Treat repo that addresses this and according to them, setting global font-face in Treat does not work as expected. I'll post this on Slack to see if anyone has the answer to this.

https://github.com/seek-oss/treat/issues/93

**Review by oddsson — COMMENTED** [review]
This looks good to me, however [this](https://github.com/island-is/island.is/pull/1206/files#r500360521) issue stands out. I don't know if we want to try to fix it here or later.

File: workspace.json (hunk 3 above)
Comment (arnifreyrs) [reply workspace.json:1869]: Then I am thinking this should be resolved in a later PR :)

**Review by oddsson — APPROVED** [review]

**Pull request merged**